### DISCUSSION TITLE
smart contract duplicate issues

### DISCUSSION THREAD

**Discussion by davedev83** [smartcontractkit/full-blockchain-solidity-course-py#660] [Q&A] [answered]
Currently on lesson 6. I'm confused about smart contracts once they've been deployed. Am I correct to assume that deploying multiple times on brownie will only "update" the original deployment (1st deploy)? or will this create duplicates? Is this a big issue?

**Comment by cromewar** [comment, marked as answer]
Hello @davedev83 when you deploy to a local blockchain (ganache) , each time you deploy you are creating a new one, but is not persistent, once the process is finished the contract disappears. Remember ganache creates a local blockchain. In the other hand when you use the same blockchain like ganache-ui you can take them as duplicates.

But if you are deploying to a live net or a testnet, you will always create a new contract with is own address, in code they could be the same (duplicates) but they will have different addresses.

**Reply by davedev83** [reply]
Thank you, that makes sense. Is this a common issue on the mainnet?

**Reply by cromewar** [reply]
Depends a lot on the situation, is not common to deploy to mainnet if you are not 100% sure, as it cost money so people are use to deploy just when they have everything figured out, if you mean someone trying to scam using a replica of a contract, this is possible yes, but not likely to happen. Anyways everyone should verify twice which contract they interact with.

**Reply by davedev83** [reply]
Thank you, very helpful.
